src/Microsoft.AspNet.Mvc.Core/ViewComponents/DefaultViewComponentHelper.cs

```diff
@@ -30,92 +33,129 @@ public void Contextualize([NotNull] ViewContext viewContext)
+        public void RenderInvoke([NotNull] string name, params object[] arguments)
         {
-            var componentType = SelectComponent(name);
-            InvokeCore(_viewContext.Writer, componentType, args);
+            var descriptor = SelectComponent(name);
+            InvokeCore(_viewContext.Writer, descriptor, arguments);
         }
 
-        public void RenderInvoke([NotNull] Type componentType, params object[] args)
+        public void RenderInvoke([NotNull] Type componentType, params object[] arguments)
         {
-            InvokeCore(_viewContext.Writer, componentType, args);
+            var descriptor = SelectComponent(componentType);
+            InvokeCore(_viewContext.Writer, descriptor, arguments);
         }
 
-        public async Task<HtmlString> InvokeAsync([NotNull] string name, params object[] args)
+        public async Task<HtmlString> InvokeAsync([NotNull] string name, params object[] arguments)
         {
-            var componentType = SelectComponent(name);
-            return await InvokeAsync(componentType, args);
+            var descriptor = SelectComponent(name);
+
+            using (var writer = new StringWriter())
+            {
+                await InvokeCoreAsync(writer, descriptor, arguments);
+                return new HtmlString(writer.ToString());
+            }
         }
 
-        public async Task<HtmlString> InvokeAsync([NotNull] Type componentType, params object[] args)
+        public async Task<HtmlString> InvokeAsync([NotNull] Type componentType, params object[] arguments)
         {
+            var descriptor = SelectComponent(componentType);
+
             using (var writer = new StringWriter())
             {
-                await InvokeCoreAsync(writer, componentType, args);
+                await InvokeCoreAsync(writer, descriptor, arguments);
                 return new HtmlString(writer.ToString());
             }
         }
 
-        public async Task RenderInvokeAsync([NotNull] string name, params object[] args)
+        public async Task RenderInvokeAsync([NotNull] string name, params object[] arguments)
         {
-            var componentType = SelectComponent(name);
-            await InvokeCoreAsync(_viewContext.Writer, componentType, args);
+            var descriptor = SelectComponent(name);
+            await InvokeCoreAsync(_viewContext.Writer, descriptor, arguments);
         }
 
-        public async Task RenderInvokeAsync([NotNull] Type componentType, params object[] args)
+        public async Task RenderInvokeAsync([NotNull] Type componentType, params object[] arguments)
         {
-            await InvokeCoreAsync(_viewContext.Writer, componentType, args);
+            var descriptor = SelectComponent(componentType);
+            await InvokeCoreAsync(_viewContext.Writer, descriptor, arguments);
         }
 
-        private Type SelectComponent([NotNull] string name)
+        private ViewComponentDescriptor SelectComponent([NotNull] string name)
```

Comment:
Why `SelectComponent` then?

I ran into a similar situation when designing `TagHelper`s and their corresponding `TagHelperDescriptor`s. Became very confusing very fast when the name "Descriptor" was intermingled with "TagHelper". 

One good example is the `TagHelperDescriptorProvider` used to have a `GetTagHelper` method which returned `TagHelperDescriptor`s. On multiple occasions it lead to unnecessary confusion when explaining how `TagHelper`s work, confusion is never bueno.